### ISSUE TITLE
'SCardTransmit() failed: 0000045D' on 5ber Ultra eUICC chip

### ISSUE THREAD

**Issue by joeyoropesa-dev** [estkme-group/lpac#157]
```console
> C:\Users\Joey Oropesa\.minilpa\windows_x86\Dir\lpac.exe version
{"type":"lpa","payload":{"code":0,"message":"success","data":"v2.1.0"}}
> C:\Users\Joey Oropesa\.minilpa\windows_x86\Dir\lpac.exe driver apdu list
{"type":"driver","payload":{"env":"DRIVER_IFID","data":[{"env":"0","name":"Generic Smart Card Reader Interface 0"}]}}
> C:\Users\Joey Oropesa\.minilpa\windows_x86\Dir\lpac.exe profile list
{"type":"lpa","payload":{"code":-1,"message":"euicc_init","data":""}}
SCardTransmit() failed: 0000045D
SCardTransmit() failed: 0000045D
```
It simply doesn't read the card at all.

**Comment by septs** [comment]
> ERROR_IO_DEVICE
>
> 1117 (0x45D)
>
> The request could not be performed because of an I/O device error.

from https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes--1000-1299-

this card has been opened by another program.

**Comment by joeyoropesa-dev** [comment]
> > ERROR_IO_DEVICE
> > 1117 (0x45D)
> > The request could not be performed because of an I/O device error.
> 
> from https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes--1000-1299-
> 
> this card has been opened by another program.

By what program - I don't have any other program that communicates with smart cards

**Comment by joeyoropesa-dev** [comment]
Also, this too:

```
SCardTransmit() failed: 8010002F
SCardTransmit() failed: 0000045D
```

**Comment by joeyoropesa-dev** [comment]
@septs So, what's the problem with this. Why it cannot just simply load. Did anyone reported these errors before about 5ber cards?

**Comment by septs** [comment]
5ber.eSIM Ultra?

**Comment by septs** [comment]
5ber.eSIM Ultra, break change ISD-R
 AID, he doesn't meet the GSMA SGP.22 standards

but this is not bug

CC @estkme

**Comment by joeyoropesa-dev** [comment]
Thank you so much for re-opening issue. If this gets implemented and support 5ber eUICC chip in the near future, this could help alot with improving support with built-in LPA modules and device could actually recognize eUICC inside 5ber and be flagged as eSIM-compatible (with EID included in *# 06#)

**Comment by septs** [comment]
No, Android use standard ISD-R AID access eUICC chip

but 5ber Ultra is changed it

so no, unless you recompile the entire AOSP

**Comment by joeyoropesa-dev** [comment]
> No, Android use standard ISD-R AID access eUICC chip
> 
> but 5ber Ultra is changed it
> 
> so no, unless you recompile the entire AOSP

So, that's why OS cannot detect eUICC - hmm... can that be changed via patching miui-framework.jar or built-in LPA to actually support it or that will require changes in low-level firmware files like kernel or something..

Since the ROM that my device is running is closed source

**Comment by AndreySV** [comment]
Could it be, why I can't open logical channel?

```
$ sudo env LPAC_APDU=at AT_DEVICE=/dev/ttyS2 AT_DEBUG=1 lpac profile list
AT_DEBUG: AT+CCHO=?
AT_DEBUG: OK
AT_DEBUG: AT+CCHC=?
AT_DEBUG: OK
AT_DEBUG: AT+CGLA=?
AT_DEBUG: OK
AT_DEBUG: AT+CCHC=1
AT_DEBUG: ERROR
AT_DEBUG: AT+CCHC=2
AT_DEBUG: ERROR
AT_DEBUG: AT+CCHC=3
AT_DEBUG: ERROR
AT_DEBUG: AT+CCHC=4
AT_DEBUG: ERROR
AT_DEBUG: AT+CCHO="A0000005591010FFFFFFFF8900000100"
AT_DEBUG: ERROR
{"type":"lpa","payload":{"code":-1,"message":"euicc_init","data":""}}
```

**Comment by joeyoropesa-dev** [comment]
Wait... will this be fixed or simply tool will accept to not be compatible with 5ber Ultra

**Comment by AndreySV** [comment]
Are 5ber Premium SIM-cards suffer from the same problem?

**Comment by joeyoropesa-dev** [comment]
> Are 5ber Premium SIM-cards suffer from the same problem?

I don't believe but you can try - maybe they started to implement this change to every new 5ber eSIM card (that includes premium and standard editions too, not just ultra but you will need to test that by yourself)

**Comment by estkme** [comment]
well, at least I know how to "support" 5ber ultra , but I have no idea wether it will make some trouble, since we are also sell eUICC cards, we have business competitive with 5ber.

**Comment by joeyoropesa-dev** [comment]
> well, at least I know how to "support" 5ber ultra , but I have no idea wether it will make some trouble, since we are also sell eUICC cards, we have business competitive with 5ber.

It would not be any issue - it could actually help 5ber.SIM company to improve support and flexibility on more devices and eSIM providers too 👍 - I would even pay extra for this kind of support to both your project and 5ber.SIM ✨

**Comment by Root-Core** [comment]
I have a similar issue with the `5ber Standard`, using Linux:

```
[manjaro-lenovo manjaro]# lpac version
{"type":"lpa","payload":{"code":0,"message":"success","data":"v2.1.0"}}
[manjaro-lenovo manjaro]# lpac driver apdu list
{"type":"driver","payload":{"env":"DRIVER_IFID","data":[{"env":"0","name":"Lenovo Integrated Smart Card Reader 00 00"}]}}
[manjaro-lenovo manjaro]# lpac profile list
{"type":"lpa","payload":{"code":-1,"message":"euicc_init","data":""}}
```
@estkme Thank you very much for your efforts, it would be greatly appreciated if this problem could be fixed. If you decide not to release anything, could you point us in the right direction?

**Comment by CoelacanthusHex** [comment]
> I have a similar issue with the `5ber Standard`, using Linux:
> 
> ```
> [manjaro-lenovo manjaro]# lpac version
> {"type":"lpa","payload":{"code":0,"message":"success","data":"v2.1.0"}}
> [manjaro-lenovo manjaro]# lpac driver apdu list
> {"type":"driver","payload":{"env":"DRIVER_IFID","data":[{"env":"0","name":"Lenovo Integrated Smart Card Reader 00 00"}]}}
> [manjaro-lenovo manjaro]# lpac profile list
> {"type":"lpa","payload":{"code":-1,"message":"euicc_init","data":""}}
> ```
> 
> @estkme Thank you very much for your efforts, it would be greatly appreciated if this problem could be fixed. If you decide not to release anything, could you point us in the right direction?

The issue about the 5ber standard edition will continue at https://github.com/estkme-group/lpac/issues/175

**Comment by estkme** [comment]
> I have a similar issue with the `5ber Standard`, using Linux:
> 
> ```
> [manjaro-lenovo manjaro]# lpac version
> {"type":"lpa","payload":{"code":0,"message":"success","data":"v2.1.0"}}
> [manjaro-lenovo manjaro]# lpac driver apdu list
> {"type":"driver","payload":{"env":"DRIVER_IFID","data":[{"env":"0","name":"Lenovo Integrated Smart Card Reader 00 00"}]}}
> [manjaro-lenovo manjaro]# lpac profile list
> {"type":"lpa","payload":{"code":-1,"message":"euicc_init","data":""}}
> ```
> 
> @estkme Thank you very much for your efforts, it would be greatly appreciated if this problem could be fixed. If you decide not to release anything, could you point us in the right direction?

https://github.com/estkme-group/lpac/commit/302984f7d59e09db4845809f2ca7cdca7b0830b1#diff-540b8e2653e840b9e2ee21f53e7b4f6c9c0ffdc0e36566824efd7273f0386073R175

**Comment by joeyoropesa-dev** [comment]
_Quick reminder for anyone who find this fixed issue_: To use custom AID feature correctly in this case for 5ber cards use this custom AID: A0000005591010FFFFFFFF8900050500
